src/core/locator/qfieldlocatorfilter.cpp

```diff
@@ -0,0 +1,147 @@
+  return filter;
+}
+
+void QFieldLocatorFilter::setSource( const QUrl &source )
+{
+  if ( mSource.path() == source.path() )
+    return;
+
+  mSource = source;
+  QUrlQuery query( mSource );
+  if ( !query.hasQueryItem( "t" ) )
+  {
+    // Bypass caching to insure updated QML content is reloaded when the plugin is loaded
+    query.addQueryItem( QStringLiteral( "t" ), QStringLiteral( "t=%1" ).arg( QDateTime::currentSecsSinceEpoch() ) );
+    mSource.setQuery( query );
+  }
+  emit sourceChanged();
+}
+
+void QFieldLocatorFilter::setLocatorBridge( LocatorModelSuperBridge *locatorBridge )
+{
+  if ( mLocatorBridge == locatorBridge )
+    return;
+
+  mLocatorBridge = locatorBridge;
+  emit locatorBridgeChanged();
+}
+
+void QFieldLocatorFilter::setName( const QString &name )
+{
+  if ( mName == name )
+    return;
+
+  mName = name;
+  emit nameChanged();
+}
+
+void QFieldLocatorFilter::setDisplayName( const QString &displayName )
+{
+  if ( mDisplayName == displayName )
+    return;
+
+  mDisplayName = displayName;
+  emit displayNameChanged();
+}
+
+void QFieldLocatorFilter::setPrefix( const QString &prefix )
+{
+  if ( mPrefix == prefix )
+    return;
+
+  mPrefix = prefix;
+  emit prefixChanged();
+}
+
+void QFieldLocatorFilter::fetchResultsEnded()
+{
+  mFetchResultsEnded = true;
+}
+
+void QFieldLocatorFilter::prepareResult( QVariant details )
+{
+  QVariantMap detailsMap = details.toMap();
+  QgsLocatorResult result;
+  result.userData() = detailsMap.value( QStringLiteral( "userData" ) );
+  result.displayString = detailsMap.value( QStringLiteral( "displayString" ), QString() ).toString();
+  result.description = detailsMap.value( QStringLiteral( "description" ), QString() ).toString();
+  result.score = detailsMap.value( QStringLiteral( "score" ), 0.5 ).toDouble();
+  result.group = detailsMap.value( QStringLiteral( "group" ), QString() ).toString();
+  result.groupScore = detailsMap.value( QStringLiteral( "groupScore" ), 0.5 ).toDouble();
+  emit resultFetched( result );
+}
+
+void QFieldLocatorFilter::fetchResults( const QString &string, const QgsLocatorContext &context, QgsFeedback *feedback )
+{
+  if ( mSource.isEmpty() )
+    return;
+
+  QQmlEngine engine;
+  QQmlComponent component( &engine );
+  component.loadUrl( mSource );
+  QObject *object = component.create();
+  if ( object )
+  {
+    mFetchResultsEnded = false;
+    connect( object, SIGNAL( prepareResult( QVariant ) ), this, SLOT( prepareResult( QVariant ) ) );
+    connect( object, SIGNAL( fetchResultsEnded() ), this, SLOT( fetchResultsEnded() ) );
+
+    QEventLoop loop;
+    connect( object, SIGNAL( fetchResultsEnded() ), &loop, SLOT( quit() ) );
+    connect( feedback, &QgsFeedback::canceled, &loop, &QEventLoop::quit );
+
+    QMetaObject::invokeMethod( object, QStringLiteral( "fetchResults" ).toStdString().c_str(), QVariant( string ), QVariant::fromValue<QgsLocatorContext>( context ) );
+    if ( !mFetchResultsEnded )
+    {
+      loop.exec();
```

Comment:
is the interface still responsive here?
i.e. what happens if there's a long-running web request (imagine timeout 60 seconds or  more)?

Reply:
Yep, the interface isn't impacted with what's happening here as this is running in a cloned filter _off_ the main QML engine / UI thread.